backend/graphql/service/product.service.ts

```diff
@@ -12,13 +12,9 @@ export class ProductService {
     input: CreateProductInput,
     context: Context
   ): Promise<Product> {
-    if (!context.user?.isAdmin) {
-      throw new NotFoundError('You are not authorized to create a product')
-    }
-
     const product = new ProductModel({
       ...input,
-      user: context.user._id
+      user: context.user?._id
```

Comment:
This should never happen.. if the AuthGuard works correctly, you will ALWAYS have a user.

You can do `context.user!._id` here.